docs/listeners/jsonapi.rst

```diff
@@ -0,0 +1,723 @@
+        },
+        "links": {
+          "self": "/currencies/1"
+        }
+      },
+      {
+        "type": "cultures",
+        "id": "2",
+        "attributes": {
+          "code": "nl-BE",
+          "name": "Dutch (Belgium)"
+        },
+        "links": {
+          "self": "/cultures/2"
+        }
+      },
+      {
+        "type": "cultures",
+        "id": "3",
+        "attributes": {
+          "code": "fr-BE",
+          "name": "French (Belgium)"
+        },
+        "links": {
+          "self": "/cultures/3"
+        }
+      }
+    ]
+  }
+
+.. note::
+
+Please note that only support for ``belongsTo`` and ``hasMany``
+relationships has been implemented.
+
+
+Configuration
+-------------
+
+The output produced by the listener is highly configurable using the Crud configuration
+options described in this section. Configure the options on the fly in your controller
+actions or enable them for all actions in your controller by adding them to the
+your controllers ``initialize()`` event like this:
+
+.. code-block:: phpinline
+
+  public function initialize()
+  {
+    parent::initialize();
+    $this->Crud->config('listeners.jsonApi.withJsonApiVersion', true);
+  }
+
+withJsonApiVersion
+^^^^^^^^^^^^^^^^^^
+
+Setting this **boolean** option to true (default: false) will make the listener add 
+the top-level ``jsonapi`` node with member node ``version`` to each response like
+shown below.
+
+.. code-block:: json
+
+  {
+    "jsonapi": {
+      "version": "1.0"
+    }
+  }  
+  
+meta
+^^^^
+
+Pass this **mixed** option (default: false) an array or hash will make the listener add the 
+the top-level ``jsonapi`` node with member node ``meta`` to each response like
+shown below. Set to ``false`` to disable the ``meta`` node.
+
+.. code-block:: json
+
+  {
+    "jsonapi": {
+      "meta": {
+        "copyright": {
+          "name": "FriendsOfCake"
+        }
+      }
+    }
+  }
+
+absoluteLinks
+^^^^^^^^^^^^^
+
+Setting this **boolean** option to true (default: false) will make the listener
+generate absolute links for the JSON API responses.
+
+debugPrettyPrint
+^^^^^^^^^^^^^^^^
+
+Setting this **boolean** option to false (default: true) will make the listener
+render non-pretty json in debug mode.
+
+
+jsonOptions
+^^^^^^^^^^^
+
+Pass this **array** option (default: empty) an array with 
+`PHP Predefined JSON Constants http://php.net/manual/en/json.constants.php`_
+to manipulate the generated json response. For example:
+
+.. code-block:: phpinline
+
+  public function initialize()
+  {
+    parent::initialize();
+    $this->Crud->config('listeners.jsonApi.jsonOptions', [
+	  JSON_HEX_QUOT,
+	  JSON_UNESCAPED_UNICODE,
+	]);
+  }
+
+include
+^^^^^^^
+
+Pass this **array** option (default: empty) an array with associated
+entities to limit the data added to the json ``included`` node.
+
+.. code-block:: phpinline
+
+  $this->Crud->config('listeners.jsonApi.include', [
+    'currency', // belongsTo relations will be singular
```

Comment:
Do these need to be Capitalised if they're an entity? Or do you mean an entity property?